File: internal/domain/kanban.go
package domain

// Inbound stages (order matters for FIFO)
const (
	InboundStageCanNhap  = "can_nhap"
	InboundStageDaLenDon = "da_len_don"
	InboundStageDaDuyet  = "da_duyet"
	InboundStageDaVeHang = "da_ve_hang"
)

// Outbound stages
const (
	OutboundStageCanDay    = "can_day"
	OutboundStageDaChotDon = "da_chot_don"
	OutboundStageDaGiao    = "da_giao"
)

// Valid stage transitions for inbound
var InboundTransitions = map[string]string{
	InboundStageCanNhap:  InboundStageDaLenDon,
	InboundStageDaLenDon: InboundStageDaDuyet,
	InboundStageDaDuyet:  InboundStageDaVeHang,
}

// Valid stage transitions for outbound
var OutboundTransitions = map[string]string{
	OutboundStageCanDay:    OutboundStageDaChotDon,
	OutboundStageDaChotDon: OutboundStageDaGiao,
}

// ValidateInboundTransition checks if the transition is valid
func ValidateInboundTransition(from, to string) bool {
	expected, ok := InboundTransitions[from]
	return ok && expected == to
}

// ValidateOutboundTransition checks if the transition is valid
func ValidateOutboundTransition(from, to string) bool {
	expected, ok := OutboundTransitions[from]
	return ok && expected == to
}

// MoveKanbanRequest is the request body for moving a kanban card
type MoveKanbanRequest struct {
	ToStage string `json:"to_stage"`
	UserID  string `json:"user_id"`
}

// CreateKanbanRequest is the request body for creating a kanban card
type CreateKanbanRequest struct {
	MaHang     string  `json:"ma_hang"`
	TenSanPham string  `json:"ten_san_pham"`
	SoLuong    float64 `json:"so_luong"`
	Note       string  `json:"note"`
}

File: web/src/types/kanban.ts
export type InboundStage =
	| "can_nhap"
	| "da_len_don"
	| "da_duyet"
	| "da_ve_hang";
export type OutboundStage = "can_day" | "da_chot_don" | "da_giao";

export interface KanbanInbound {
	id: number;
	ma_hang: string;
	ten_san_pham: string;
	so_luong: number;
	stage: InboundStage;
	note: string;
	created_at: string;
	updated_at: string;
}

export interface KanbanOutbound {
	id: number;
	ma_hang: string;
	ten_san_pham: string;
	so_luong: number;
	stage: OutboundStage;
	note: string;
	created_at: string;
	updated_at: string;
}

export interface CreateKanbanRequest {
	ma_hang: string;
	ten_san_pham: string;
	so_luong: number;
	note: string;
}

export interface MoveKanbanRequest {
	to_stage: string;
	user_id: string;
}

export const INBOUND_STAGES: { key: InboundStage; label: string }[] = [
	{ key: "can_nhap", label: "Cần nhập" },
	{ key: "da_len_don", label: "Đã lên đơn" },
	{ key: "da_duyet", label: "Đã duyệt" },
	{ key: "da_ve_hang", label: "Đã về hàng" },
];

export const OUTBOUND_STAGES: { key: OutboundStage; label: string }[] = [
	{ key: "can_day", label: "Cần đẩy" },
	{ key: "da_chot_don", label: "Đã chốt đơn" },
	{ key: "da_giao", label: "Đã giao" },
];
